### PR TITLE
EN-2197: add docker images for ruby 2.3 and rails 4.2

### PR DESCRIPTION
**Pull request by dbussone**

### CODE REVIEW

File: rails4-deps/Gemfile.lock [hunk 1]
```diff
@@ -0,0 +1,396 @@
+GIT
```
Comment (michaelb990) [inline rails4-deps/Gemfile.lock]: Do we want this committed?

File: ruby/2.3/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  DEBIAN_FRONTEND=noninteractive apt-get install -y ruby2.3 && \
+  DEBIAN_FRONTEND=noninteractive apt-get purge -y --auto-remove software-properties-common && \
+  rm -rf /var/lib/apt/lists/*
+
+# skip installing gem documentation
+RUN echo 'gem: --no-rdoc --no-ri --no-document' >> "/etc/gemrc" && \
+  gem install bundler
+
+# LABEL must be last for proper base image discoverability
+LABEL repository.socrata/ruby:2.3=""
```
Comment (dbussone) [inline ruby/2.3/Dockerfile]: should this colon be there?

File: rails4-deps/Gemfile.lock (hunk 1 above)
Comment (dbussone) [inline rails4-deps/Gemfile.lock]: i assumed so. the rails3-deps contains a Gemfile.lock

File: ruby/2.3/Dockerfile (hunk 2 above)
Comment (michaelb990) [inline ruby/2.3/Dockerfile]: This should be as follows to match the name of the docker image (socrata/ruby2.3)
```dockerfile
LABEL repository.socrata/ruby2.3=""
```

File: rails4-deps/Gemfile.lock (hunk 1 above)
Comment (michaelb990) [inline rails4-deps/Gemfile.lock]: :+1:

**Comment by michaelb990** [comment]
lgtm

**Pull request merged**